pants

```diff
@@ -13,11 +13,15 @@
 
 set -eou pipefail
 
-PYTHON=${PYTHON:-$(which python2.7)}
-
 PANTS_HOME="${PANTS_HOME:-${XDG_CACHE_HOME:-$HOME/.cache}/pants/setup}"
 PANTS_BOOTSTRAP="${PANTS_HOME}/bootstrap-$(uname -s)-$(uname -m)"
 
+interpreter_version=$(
+  grep -E "^[[:space:]]*pants_engine_python_version" pants.ini 2>/dev/null | \
+    cut -f2 -d: | tr -d " "
+) || interpreter_version=2.7
+PYTHON=${PYTHON:-$(which python${interpreter_version})}
```

Comment:
Is it worth checking that this resulted in a non-empty value, and erroring if not? Or does the `venv` creation result in a useful error message if your choice isn't installed?